syntax = "proto3";

package order_queue;

service OrderQueueService {
  rpc Enqueue(EnqueueRequest) returns (EnqueueResponse) {}
  rpc Dequeue(DequeueRequest) returns (DequeueResponse) {}
}

message EnqueueRequest {
  string orderId = 1;
  repeated Item items = 2;
}

message Item {
  string name = 1;
  int32 quantity = 2;
}

message EnqueueResponse {
  bool success = 1;
}

message DequeueRequest {
}

message DequeueResponse {
  string orderId = 1;
  repeated Item items = 2;
  bool success = 3;
}

